SyncAdapterLab/app/src/main/java/net/serkanbal/syncadapterlab/MainActivity.java

```diff
@@ -0,0 +1,98 @@
+import android.support.v7.widget.LinearLayoutManager;
+import android.support.v7.widget.RecyclerView;
+
+import java.util.ArrayList;
+
+public class MainActivity extends AppCompatActivity implements LoaderManager.LoaderCallbacks<Cursor>{
+    public static final String ACCOUNT_TYPE = "example.com";
+    public static final String ACCOUNT = "default_account";
+    public static final int PORTFOLIO_LOADER = 0;
+    private RecyclerView mPortfolioRecyclerView;
+    private PortfolioRecyclerAdapter mAdapter;
+
+    Account mAccount;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+
+        mAccount = createSyncAccount(this);
+
+        mPortfolioRecyclerView = (RecyclerView) findViewById(R.id.recyclerview);
+        mPortfolioRecyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL,false));
+
+        mAdapter = new PortfolioRecyclerAdapter(new ArrayList<CompanyInfo>());
+        mPortfolioRecyclerView.setAdapter(mAdapter);
+
+        getSupportLoaderManager().initLoader(PORTFOLIO_LOADER, Bundle.EMPTY, this);
+
+        //Manual sync
+//        Bundle settingsBundle = new Bundle();
+//        settingsBundle.putBoolean(
+//                ContentResolver.SYNC_EXTRAS_MANUAL, true);
+//        settingsBundle.putBoolean(
+//                ContentResolver.SYNC_EXTRAS_EXPEDITED, true);
+//        ContentResolver.requestSync(mAccount, StockPortfolioContract.AUTHORITY, settingsBundle);
+
+        //Periodic sync
+        ContentResolver.setSyncAutomatically(mAccount, StockPortfolioContract.AUTHORITY,true);
+        ContentResolver.addPeriodicSync(
+                mAccount,
+                StockPortfolioContract.AUTHORITY,
+                Bundle.EMPTY,
+                60);
+
+    }
+
+    public static Account createSyncAccount(Context context) {
+        Account newAccount = new Account(
+                ACCOUNT, ACCOUNT_TYPE);
+
+        AccountManager accountManager =
+                (AccountManager) context.getSystemService(
+                        ACCOUNT_SERVICE);
+
+        if (accountManager.addAccountExplicitly(newAccount, null, null)) {
+        } else {
+        }
+        return newAccount;
+    }
+
+    @Override
+    public Loader<Cursor> onCreateLoader(int id, Bundle args) {
+        switch (id) {
+            case PORTFOLIO_LOADER:
+                return new CursorLoader(this,
+                        StockPortfolioContract.Stocks.CONTENT_URI,
+                        null,
+                        null,
+                        null,
+                        null);
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public void onLoadFinished(Loader<Cursor> loader, Cursor data) {
+        mAdapter.swapData(data);
+    }
+
+    @Override
+    public void onLoaderReset(Loader<Cursor> loader) {
+        //
```

Comment:
Reset indicates that the data is no longer valid, so rather than do nothing, you should update the RecyclerView with null or an empty list